covenant/covenant.go

```diff
@@ -436,30 +439,79 @@ func (ce *CovenantEmulator) delegationsToBatches(dels []*types.Delegation) [][]*
 	return batches
 }
 
-func RemoveAlreadySigned(localKey *btcec.PublicKey, dels []*types.Delegation) []*types.Delegation {
-	sanitized := make([]*types.Delegation, 0, len(dels))
-	localKeyBytes := schnorr.SerializePubKey(localKey)
+// IsKeyInCommittee returns true if the covenant serialized public key is in the covenant committee of the
+// parameter in which the BTC delegation was included.
+func IsKeyInCommittee(paramCache ParamsGetter, covenantSerializedPk []byte, del *types.Delegation) (bool, error) {
+	stkParams, err := paramCache.Get(del.ParamsVersion)
+	if err != nil {
+		return false, err
```

Comment:
sorry, for not being super clear, what I meant is to here return the `fmt.Errorf"msg: %w", err)` in case of error